plugins/pagerduty/src/components/types.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DiscoveryApi } from '@backstage/core';
+
+export type Incident = {
+  id: string;
+  title: string;
+  status: string;
+  html_url: string;
+  assignments: [
+    {
+      assignee: Assignee;
+    },
+  ];
+  serviceId: string;
+  created_at: string;
+};
+
+export type Service = {
+  id: string;
+  name: string;
+  html_url: string;
+  integrationKey: string;
+  escalation_policy: {
+    id: string;
+    user: User;
+  };
+};
+
+export type OnCall = {
+  user: User;
+};
+
+export type Assignee = {
+  id: string;
+  summary: string;
+  html_url: string;
+};
+
+export type User = {
+  id: string;
+  summary: string;
+  email: string;
+  html_url: string;
+  name: string;
+};
+
+export type ServicesResponse = {
+  services: Service[];
+};
+
+export type IncidentsResponse = {
+  incidents: Incident[];
+};
+
+export type OnCallsResponse = {
+  oncalls: OnCall[];
+};
+
+export type RequestOptions = {
+  method: string;
+  headers: HeadersInit;
+  body?: BodyInit;
+};
+
+export type ClientApiConfig = {
```

Comment:
I would probably just inline this in the API client itself. Either that or convert to use the new config schema API. For example: https://github.com/backstage/backstage/pull/3427/files

Reply:
We will save this for next iteration.